### DISCUSSION TITLE
Cloudflare tunnels

### DISCUSSION THREAD

**Discussion by Shootify** [tteck/Proxmox#1611] [General]
HI guys, anyone here running this scripted awesome apps with cloudflare, I just deploy Vaultwarden on my Lab using the script provided and seems to be working fine... anyone else?

**Comment by Eddieargenal** [comment]
downloaded and installed the container using the helper's script, but now looking for guidance on how to set it up. Could you point me in the right direction?

**Reply by MivraMe** [reply]
Did you found how ?

**Reply by tteck** [reply]
This resource might be useful: https://www.youtube.com/watch?v=EOcwVjdCAEc. The Cloudflared script isn't using Docker,  we're opting for a Cloudflared (Debian) LXC solution instead.

**Reply by tteck** [reply]
Also if using it for Home Assistant, don't forget to set `http:` in the `configuration.yaml`

```
http:
  use_x_forwarded_for: true
  trusted_proxies:
    - <Cloudflared LXC IP>
```

**Reply by Eddieargenal** [reply]
Man, you see awesome! Thanks for all your support!

**Comment by DrSpaldo** [comment]
Thanks for the script (as always!) 

I've also just installed the Cloudflared LXC using the helper script and trying to work out how to set it up... 

Where do I put the token? where do I put the external subdomain -> internal IP service links like in the HA addon?

**Reply by DrSpaldo** [reply]
I might be getting somewhere on this. Looks like loading the console with the Cloudflared LXC and running this command linked my tunnel to the LXC:

sudo cloudflared service install YOURTOKEN

**Reply by DrSpaldo** [reply]
So it looks like once you have linked the service to that tunnel/token, then you can follow the instructions on the video posted above around the 6:10 mark. Just by adding the public hostnames to the tunnel using your internal IP/Port. I am not sure if it is worth adding this into the docs?

**Reply by syl779** [reply]
Thanks. Typing:

cloudflared service install YOURTOKEN

in the container's console is the missing part of the puzzle.

**Reply by DrSpaldo** [reply]
@tteck maybe this can be added to the script/install page?

**Comment by danielgoz** [comment]
Hi All, not sure if this is the best place to ask but....
How should I do periodic updates on this container?
I found: apt-get update && apt-get upgrade cloudflared
It looks like it updated the whole system (LXC container) including cloudflared.
How can I check cloudflare version?
I also noticed that you can't do it via proxmox console, you need to SSH into the container.

Thanks
Dan

**Reply by syl779** [reply]
There’s a script for that (run it on the Proxmox VE shell).

[LXC updater](https://tteck.github.io/Proxmox/#proxmox-ve-lxc-updater
)
`bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/misc/update-lxcs.sh)"
`

There’s even one to add it to the crontab.

**Reply by danielgoz** [reply]
Noice... Thanks!

**Comment by tteck** [comment]
```
root@cloudflared:~# cloudflared --version
cloudflared version 2024.10.0 (built 2024-10-10-0949 UTC)
```
